From owner-linux-pci@atrey.karlin.mff.cuni.cz Sun Mar 30 15:41:45 2008
From: Adrian Bunk <bunk@kernel.org>
Date: Mon, 31 Mar 2008 01:41:01 +0300
Subject: PCI: aerdrv_acpi.c: remove unneeded NULL check
To: Greg KH <greg@kroah.com>
Cc: Andrew Patterson <andrew.patterson@hp.com>
Message-ID: <20080330224101.GO28445@cs181133002.pp.htv.fi>
Content-Disposition: inline


There's no reason for checking pdev->bus for being NULL here (and we'd
anyway Oops 3 lines below if it was).

Signed-off-by: Adrian Bunk <bunk@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/pcie/aer/aerdrv_acpi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/pci/pcie/aer/aerdrv_acpi.c
+++ b/drivers/pci/pcie/aer/aerdrv_acpi.c
@@ -37,7 +37,7 @@ int aer_osc_setup(struct pcie_device *pc
 		return -1;
 
 	/* Find root host bridge */
-	while (pdev->bus && pdev->bus->self)
+	while (pdev->bus->self)
 		pdev = pdev->bus->self;
 	handle = acpi_get_pci_rootbridge_handle(
 		pci_domain_nr(pdev->bus), pdev->bus->number);
